graphrag_sdk/models/litellm.py

```diff
@@ -0,0 +1,283 @@
+        
+        elif self.provider == "ollama":
+            self.ollama_client = Client()
+            self.check_and_pull_model()
+            
+    def check_and_pull_model(self) -> None:
+        """
+        Checks if the specified model is available locally, and pulls it if not.
+
+        Logs:
+            - Info: If the model is already available or after successfully pulling the model.
+            - Error: If there is a failure in pulling the model.
+
+        Raises:
+            Exception: If there is an error during the model pull process.
+        """
+        # Get the list of available models
+        response = self.ollama_client.list()  # This returns a dictionary
+        available_models = [model['name'] for model in response['models']]  # Extract model names
+
+        # Check if the model is already pulled
+        if self.model_name in available_models:
+            logger.info(f"The model '{self.model_name}' is already available.")
+        else:
+            logger.info(f"Pulling the model '{self.model_name}'...")
+            try:
+                self.ollama_client.pull(self.model_name)  # Pull the model
+                logger.info(f"Model '{self.model_name}' pulled successfully.")
+            except Exception as e:
+                logger.error(f"Failed to pull the model '{self.model_name}': {e}")
+
+    def with_system_instruction(self, system_instruction: str) -> "GenerativeModel":
+        """
+        Set or update the system instruction for new model instance.
+
+        Args:
+            system_instruction (str): Instruction for guiding the model's behavior.
+        
+        Returns:
+            GenerativeModel: The updated model instance.
+        """
+        self.system_instruction = system_instruction
+        return self
+
+    def start_chat(self, args: Optional[dict] = None) -> GenerativeModelChatSession:
+        """
+        Start a new chat session.
+
+        Args:
+            args (Optional[dict]): Additional arguments for the chat session.
+
+        Returns:
+            GenerativeModelChatSession: A new instance of the chat session.
+        """
+        return LiteModelChatSession(self, args)
+
+    def parse_generate_content_response(self, response: any) -> GenerationResponse:
+        """
+        Parse the model's response and extract content for the user.
+
+        Args:
+            response (any): The raw response from the model.
+
+        Returns:
+            GenerationResponse: Parsed response containing the generated text.
+        """
+        return GenerationResponse(
+            text=response.choices[0].message.content,
+            finish_reason=(
+                FinishReason.STOP
+                if response.choices[0].finish_reason == "stop"
+                else (
+                    FinishReason.MAX_TOKENS
+                    if response.choices[0].finish_reason == "length"
+                    else FinishReason.OTHER
```

Comment:
shouldn't there be a `FinishReason` which indicate success,
both `stop` and `max_tokens` indicate some sort of an error, `other` is unclear.

Reply:
Currently we only track output errors, why do you think we need an indication of success?